WordPress/src/main/java/org/wordpress/android/ui/posts/editor/StoriesEventListener.kt

```diff
@@ -0,0 +1,158 @@
+package org.wordpress.android.ui.posts.editor
+
+import android.net.Uri
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.Lifecycle.State.CREATED
+import androidx.lifecycle.LifecycleObserver
+import androidx.lifecycle.OnLifecycleEvent
+import com.wordpress.stories.compose.frame.StorySaveEvents.FrameSaveCompleted
+import com.wordpress.stories.compose.frame.StorySaveEvents.FrameSaveFailed
+import com.wordpress.stories.compose.frame.StorySaveEvents.FrameSaveProgress
+import com.wordpress.stories.compose.frame.StorySaveEvents.FrameSaveStart
+import com.wordpress.stories.compose.frame.StorySaveEvents.StorySaveResult
+import org.greenrobot.eventbus.EventBus
+import org.greenrobot.eventbus.Subscribe
+import org.greenrobot.eventbus.ThreadMode
+import org.wordpress.android.editor.gutenberg.StorySaveMediaListener
+import org.wordpress.android.fluxc.Dispatcher
+import org.wordpress.android.fluxc.model.MediaModel
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.store.MediaStore
+import org.wordpress.android.ui.stories.SaveStoryGutenbergBlockUseCase.Companion.TEMPORARY_ID_PREFIX
+import org.wordpress.android.ui.stories.StoryRepositoryWrapper
+import org.wordpress.android.ui.stories.media.StoryMediaSaveUploadBridge.StoryFrameMediaModelCreatedEvent
+import org.wordpress.android.util.FluxCUtils
+import org.wordpress.android.util.helpers.MediaFile
+import javax.inject.Inject
+
+class StoriesEventListener @Inject constructor(
+    private val dispatcher: Dispatcher,
+    private val mediaStore: MediaStore,
+    private val storyRepositoryWrapper: StoryRepositoryWrapper
+) : LifecycleObserver {
+    private lateinit var lifecycle: Lifecycle
+    private lateinit var site: SiteModel
+    private var storySaveMediaListener: StorySaveMediaListener? = null
+
+    @OnLifecycleEvent(Lifecycle.Event.ON_CREATE)
+    private fun onCreate() {
+        dispatcher.register(this)
+        EventBus.getDefault().register(this)
```

Comment:
Hi @mzorz I think these could be injected as well. There's probably an `EventBusWrapper`.

Reply:
👍 addressed in d065c92af4